### DISCUSSION TITLE
leptos_dom::html::HtmlElement::dyn_classes How to use?

### DISCUSSION THREAD

**Discussion by git-lloyd** [leptos-rs/leptos#1264] [Q&A] [answered]
when function input parameter  classes （signal） from vec!["aaa".into()]  -> vec!["bbb".into()]
the div class is empty.

leptos = "0.4"

#[component]
pub fn content(
    cx: Scope,
    children: Children,
    #[prop(optional, into)] classes: MaybeSignal<Vec<String>>,
    #[prop(optional, into)] tag: String,
) -> impl IntoView {
    let mut tag = tag;
    if tag.is_empty() {
        tag = "div".into();
    }
    let element = leptos::html::custom(cx, leptos_dom::html::Custom::new(tag));
    let element = element.child(children(cx));
    let classes = move || {
        let mut classes = classes.get();
        classes.push("content".into());
        classes
    };
    element.dyn_classes(classes)
}

**Comment by gbj** [comment, marked as answer]
Thanks, it turns out it was just bugged. PR to fix it here https://github.com/leptos-rs/leptos/pull/1265

**Reply by git-lloyd** [reply]
Thank you very much!  
at leptos_dom::html::HtmlElement.dyn_classes:line752
panic!("failed to add class \ `{prev_class}`, error: {err:#?}") should be panic!("failed to remove class \ `{prev_class}`, error: {err:#?}") ？

code snippet：
// Remove old classes
for prev_class in old_classes.clone() {
    if !new_classes.iter().any(|c| c == &prev_class) {
        class_list.remove_1(prev_class).unwrap_or_else(
            |err| {
                panic!(
                    "failed to add class \
                     `{prev_class}`, error: {err:#?}"
                )
            },
        );
    }
}
